src/main/java/co/cask/bigquery/BigQuerySink.java

```diff
@@ -243,14 +262,38 @@ public void transform(StructuredRecord input, Emitter<KeyValue<JsonObject, NullW
     emitter.emit(new KeyValue<>(object, NullWritable.get()));
   }
 
-  private void decodeSimpleTypes(JsonObject json, String name, Object object, Schema schema) throws RecordConvertorException {
-    Schema.Type type = schema.getType();
+  private void decodeSimpleTypes(JsonObject json, String name, StructuredRecord input, Schema schema)
+    throws RecordConverterException {
+    Object object = input.get(name);
 
     if (object == null) {
       json.add(name, JsonNull.INSTANCE);
       return;
     }
 
+    Schema.LogicalType logicalType = schema.getLogicalType();
```

Comment:
this doesn't look like it handles nullable types.

Reply:
ok I see it's because it's done by the caller... this class needs a lot of cleanup (not in this PR).

Reply:
yea both `BigQuerySource` and `BigQuerySink` requires clean up.